src/sentry/search/snuba/executors.py

```diff
@@ -71,134 +40,184 @@ def get_search_filter(search_filters, name, operator):
     return found_val
 
 
+@six.add_metaclass(ABCMeta)
 class AbstractQueryExecutor:
     """This class serves as a template for Query Executors.
     We subclass it in order to implement query methods (we use it to implement two classes: joined Postgres+Snuba queries, and Snuba only queries)
     It's used to keep the query logic out of the actual search backend,
     which can now just build query parameters and use the appropriate query executor to run the query
     """
 
-    def query(
-        self,
-        projects,
-        retention_window_start,
-        group_queryset,
-        environments,
-        sort_by,
-        limit,
-        cursor,
-        count_hits,
-        paginator_options,
-        search_filters,
-        date_from,
-        date_to,
-    ):
-        raise NotImplementedError
+    EMPTY_RESULT = Paginator(Group.objects.none()).get_result()
+    TABLE_ALIAS = ""
 
+    @abstractmethod
+    def query(self, *args, **kwargs):
+        raise NotImplementedError
 
-def snuba_search(
-    start,
-    end,
-    project_ids,
-    environment_ids,
-    sort_field,
-    cursor=None,
-    candidate_ids=None,
-    limit=None,
-    offset=0,
-    get_sample=False,
-    search_filters=None,
-):
-    """
-    This function doesn't strictly benefit from or require being pulled out of the main
-    query method above, but the query method is already large and this function at least
-    extracts most of the Snuba-specific logic.
+    @abstractmethod
+    def calculate_hits(self, *args, **kwargs):
+        raise NotImplementedError
```

Comment:
See my previous comment on using *args and **kwargs. 
In practice, there is only one place that make a call to these methods, so the parameters passed will always be the same, no matter what the implementation uses. Thus not declaring them does not buy you anything.
On the contrary, it defies the purpose of defining a contract between the abstract class and the implementations. IT would not be clear anymore what the abstract class is supposed to provide and what an implementation is supposed to work with.

Reply:
calculate_hits is actually implemented slightly differently in the subclass executors. The only difference is that one accepts group_ids (the postgres one) and the snuba one doesn't take this parameter. Would you still suggest defining the arguments in the abstract method in this case?

Reply:
Yes, because the place where the call is made does not know this difference and should not know this difference.

Reply:
I removed the abstract definition of calculate_hits